geos/entrypoints.go

```diff
@@ -4,6 +4,17 @@ package geos
 #cgo LDFLAGS: -lgeos_c
 #cgo CFLAGS: -Wall
 #include "geos_c.h"
+
+#define MAKE_VALID_MIN_VERSION "3.8.0"
+#define MAKE_VALID_MISSING ( \
+	GEOS_VERSION_MAJOR < 3 || \
+	(GEOS_VERSION_MAJOR == 3 && GEOS_VERSION_MINOR < 8) \
+)
+#if MAKE_VALID_MISSING
+// This stub implementation always fails:
+GEOSGeometry *GEOSMakeValid_r(GEOSContextHandle_t handle, const GEOSGeometry* g) { return NULL; }
```

Comment:
Is this needed to prevent a compile-time error?

Reply:
Yes, that's correct. Prior versions of the GEOS library don't define the symbol at all.